src/main/java/com/linkedin/kafka/clients/consumer/LiKafkaConsumerImpl.java

```diff
@@ -221,6 +227,14 @@ public void unsubscribe() {
 
   @Override
   public ConsumerRecords<K, V> poll(long timeout) {
+    if (_lastProcessedResult != null && _lastProcessedResult.exception() != null) {
+      for (Map.Entry<TopicPartition, Long> entry : _lastProcessedResult.resumeOffsets().entrySet()) {
+        _kafkaConsumer.seek(entry.getKey(), entry.getValue());
+      }
+      RuntimeException e = _lastProcessedResult.exception();
+      _lastProcessedResult = null;
+      throw e;
```

Comment:
No, the point of the patch is double folded. If the `skip.record.on.exception` is set to true. We do not throw exception. Otherwise, we throw exception, but user can choose to catch it and continue to call poll() so the exception will be skipped.